server/dals/expression-dal.js

```diff
@@ -107,16 +130,34 @@ let convertExpressionToJSON = function (totalOutput, dataset, timePoints, allGen
     return JSONOutput;
 };
 
+const convertProductionDegradationRateToJSON = (totalOutput, rateType) => {
+    const JSONOutput = {
+    };
+    for (let gene of totalOutput) {
+        JSONOutput[gene.display_gene_id] = gene[rateType];
+    }
+    return JSONOutput;
+};
+
 module.exports = {
     queryExpressionDatabase: function (req, res) {
-        return sequelize.query(buildExpressionQuery(req.query.dataset, req.query.timepoints, req.query.genes),
+        return sequelize.query(buildExpressionQuery(req.query),
                 { type: sequelize.QueryTypes.SELECT })
                     .then(function (stdname) {
-                        let dataset = req.query.dataset;
-                        let geneList = req.query.genes.split(",");
-                        let response = convertExpressionToJSON(
-                            stdname, dataset, expressionTimepointsByDataset[dataset], geneList);
-                        return res.send(response);
+                        const type = req.query.type;
+                        if (type === "DegradationRates") {
+                            return res.send(convertProductionDegradationRateToJSON(stdname, "degradation_rate"));
+
+                        } else if (type === "ProductionRates") {
+                            return res.send(convertProductionDegradationRateToJSON(stdname, "production_rate"));
+
+                        } else if (type === undefined) {
+                            let dataset = req.query.dataset;
+                            let geneList = req.query.genes.split(",");
+                            let response = convertExpressionToJSON(
+                                stdname, dataset, expressionTimepointsByDataset[dataset], geneList);
+                            return res.send(response);
+                        }
```

Comment:
Hmmm second time I’m seeing this pattern…maybe we should consider refactoring to data-driven or else we might never get around to it 😅

Reply:
I changed the if else and the above switch statements for these functions, however there are still a lot of switch statements and if else in the code. Should I do a refactor for all instances of this?